Sources/PointFree/Account/Change.swift

```diff
@@ -0,0 +1,392 @@
+      .run
+      .flatMap(
+        either(
+          const(
+            conn |> redirect(
+              to: .account(.subscription(.change(.show))),
+              headersMiddleware: flash(
+                .error,
+                """
+                We couldn’t modify your subscription at this time. Please try again or contact
+                <support@pointfree.co>.
+                """
+              )
+            )
+          )
+        ) { _ in
+          // TODO: Send email?
+
+          return conn |> redirect(
+            to: .account(.index),
+            headersMiddleware: flash(.notice, "We’ve modified your subscription.")
+          )
+        }
+    )
+}
+
+func requireActiveSubscription<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T3<Stripe.Subscription, Database.User, A>, Data>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, T3<Stripe.Subscription, Database.User, A>, Data> {
+
+    return filter(
+      get1 >>> (^\.status == .active),
+      or: redirect(
+        to: .pricing(nil, expand: nil),
+        headersMiddleware: flash(
+          .error,
+          "You don’t have an active subscription. Would you like to subscribe?"
+        )
+      )
+      )
+      <| middleware
+}
+
+private func requireValidSeating(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, Tuple4<Stripe.Subscription, Database.User, Int, Pricing>, Data>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, Tuple4<Stripe.Subscription, Database.User, Int, Pricing>, Data> {
+
+    return filter(
+      seatsAvailable,
+      or: redirect(
+        to: .account(.index),
+        headersMiddleware: flash(
+          .error,
+          "We can’t reduce the number of seats below the number that are active."
+        )
+      )
+      )
+      <| middleware
+}
+
+private func seatsAvailable(_ data: Tuple4<Stripe.Subscription, Database.User, Int, Pricing>) -> Bool {
+  let (_, _, seatsTaken, pricing) = lower(data)
+
+  return pricing.quantity >= seatsTaken
+}
+
+private let extraStyles =
+  ((input & .pseudo(.checked) ~ .star) > .star) % (
+    color(Colors.black)
+      <> fontWeight(.bold)
+    )
+    <> extraSpinnerStyles
+
+private func fetchSeatsTaken<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T3<Database.User, Int, A>, Data>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, A>, Data> {
+
+    return { conn -> IO<Conn<ResponseEnded, Data>> in
+      let user = conn.data.first
+
+      let invitesAndTeammates = sequence([
+        parallel(AppEnvironment.current.database.fetchTeamInvites(user.id).run)
+          .map { $0.right?.count ?? 0 },
+        parallel(AppEnvironment.current.database.fetchSubscriptionTeammatesByOwnerId(user.id).run)
+          .map { $0.right?.count ?? 0 }
+        ])
+
+      return invitesAndTeammates
+        .sequential
+        .flatMap { middleware(conn.map(const(user .*. $0.reduce(1, +) .*. conn.data.second))) }
```

Comment:
we could cook up an `insert1`, `insert2`, ... family of operators for inserting values into tuples:

```swift
func insert1<A, B, Z>(_ a: A) -> (T2<B, Z>) -> T3<A, B, Z> {
  return { t in 
    return a .*. t.first .*. t.second
  }
}
```